src/ctia/schemas/graphql/pagination.clj

```diff
@@ -0,0 +1,212 @@
+  (:import graphql.Scalars))
+
+(def PageInfo
+  (g/new-object
+   "PageInfo"
+   ""
+   []
+   {:hasNextPage {:type (g/non-null Scalars/GraphQLBoolean)}
+    :hasPreviousPage {:type (g/non-null Scalars/GraphQLBoolean)}
+    :startCursor {:type Scalars/GraphQLString}
+    :endCursor {:type Scalars/GraphQLString}}))
+
+(def connection-arguments
+  {:after {:type Scalars/GraphQLString}
+   :first {:type Scalars/GraphQLInt
+           :default 50}
+   :before {:type Scalars/GraphQLString}
+   :last {:type Scalars/GraphQLInt}})
+
+(defn new-edge
+  [node-type edge-name]
+  (g/new-object
+   edge-name
+   "An edge in a connection."
+   []
+   {:node {:type node-type}
+    :cursor {:type (g/non-null Scalars/GraphQLString)}}))
+
+(defn new-connection
+  [node-type]
+  (let [type-name (str/capitalize (.getName node-type))
+        connection-name (str type-name
+                             "Connection")
+        edge-name (str type-name
+                       "Edge")]
+    (g/new-object
+     connection-name
+     (str "A connection to a list of " type-name)
+     []
+     {:pageInfo {:type (g/non-null PageInfo)}
+      :totalCount {:type Scalars/GraphQLInt}
+      :edges {:type (g/list-type (new-edge node-type
+                                           edge-name))}
+      :nodes {:type (g/list-type node-type)}})))
+
+;;------- Limit/Offset with opaque cursor
+;; See : https://github.com/darthtrevino/relay-cursor-paging/blob/master/src/getPagingParameters.ts
+
+;; |--C0--|--C1--|--C2--|
+;; C0: offset=0, C1: offset=1
+;; To select the node corresponding to the C1 cursor:
+;; - {:after C0 :first 1} -> {:offset 1 :limit 1}
+;; - {:before C2 :last 1} -> {:offset 1 :limit 1}
+
+(def default-limit 50)
+
+(def Cursor s/Str)
+
+(s/defschema PagingDirection
+  {:forward-paging? s/Bool
+   :backward-paging? s/Bool})
+
+(s/defschema ConnectionParams
+  (st/merge
+   {s/Keyword s/Any}
+   (st/optional-keys
+    {:first (s/maybe s/Int)
+     :last (s/maybe s/Int)
+     :after (s/maybe Cursor)
+     :before (s/maybe Cursor)})))
+
+(s/defschema PagingParams
+  (st/merge
+   {:limit s/Int
+    :offset s/Int}
+   PagingDirection))
+
+(s/defschema Edge
+  {:cursor Cursor
+   :node s/Any})
+
+(s/defschema Connection
+  {:pageInfo {:hasNextPage s/Bool
+              :hasPreviousPage s/Bool
+              :startCursor (s/maybe Cursor)
+              :endCursor (s/maybe Cursor)}
+   :totalCount s/Int
+   :edges (s/maybe [Edge])
+   :nodes (s/maybe [s/Any])})
+
+(s/defschema Result
+  {:data [s/Any]
+   :paging {s/Any s/Any}})
+
+(s/defn serialize-cursor :- (s/maybe Cursor)
+  [offset :- (s/maybe s/Int)]
+  (when offset
+    (b64/encode (str offset))))
+
+(s/defn unserialize-cursor :- (s/maybe s/Int)
+  [cursor :- (s/maybe Cursor)]
+  (when cursor
+    (try
+      (-> cursor
+          b64/decode
+          (Integer/parseInt))
+      (catch Exception e
+        (log/warn e "Unable to unserialize cursor")
+        0))))
+
+(s/defn validate-paging :- PagingDirection
```

Comment:
Would it be possible to return something like {:error s/Str :data PaginDirection} instead?
So we'll be able to use return value instead of thrown exceptions.